Lib/test/test_ntpath.py

```diff
@@ -889,6 +890,26 @@ def test_isjunction(self):
                 self.assertFalse(ntpath.isjunction('tmpdir'))
                 self.assertPathEqual(ntpath.realpath('testjunc'), ntpath.realpath('tmpdir'))
 
+    @unittest.skipIf(sys.platform != 'win32', "drive letters are a windows concept")
+    def test_isfile_driveletter(self):
+        current_drive = "\\\\.\\" + os.path.splitdrive(os.path.abspath(__file__))[0]
+        self.assertFalse(os.path.isfile(current_drive))
```

Comment:
The test could be run on a UNC path, such as "\\\\localhost\\C$\\Source\cpython". In the latter case, the 'drive' of `__file__` would be "\\\\localhost\\C$". I'd use the `SystemDrive` environment variable, and skip the test if it doesn't exist or isn't a drive letter.

```suggestion
        drive = os.environ.get('SystemDrive')
        if drive is None or len(drive) != 2 or drive[1] != ':':
            raise unittest.SkipTest('SystemDrive is not defined or malformed')
        self.assertFalse(os.path.isfile('\\\\.\\' + drive))
```